File: src/main/resources/static/app/classes/Track.ts
/**
 * Class that represent the information related to MiausicBox tracks
 * @class Info
 */

export class Track{

  /* Attributes */
  private _id: number;
  private _name: string;
  private _group: string;
  private _link: string;

  /* Constructor */
  constructor(id:number, name:string, group: string, link:string) {
    this._id = id;
    this._name = name;
    this._group = group;
    this._link = link;
  }

  /* Getters & Setters */
  get name():string {
    return this._name;
  }

  get group():string {
    return this._group;
  }

  get link():string {
    return this._link;
  }

  set name(name:string) {
    this._name = name;
  }

  set group(group:string) {
    this._group =  group;
  }

  set link(link:string) {
    this._link = link;
  }

  /** Return if a Track is equal to other
   *  @method equals
   *  @param {object} object
   */
  equals(object:any) {
    if (!(object instanceof Track)) {
      return false;
    } else {
      var track:Track = object;
      return (track.name == this.name &&
              track._group == this.group);
    }
  }

}

File: src/main/java/giraffe/miausicbox/model/Track.java
package giraffe.miausicbox.model;

import java.util.Objects;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Track {

	public interface Basic {}
	
	// Attributes
	@JsonView(Basic.class)
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private long id;
	
	@JsonView(Basic.class)
	private String name;
	
	@JsonView(Basic.class)
	private String band;
	
	@JsonView(Basic.class)
	private String link;
	
	// Constructor
	public Track() {}
	
	public Track(
			String name,
			String band,
			String link
			) {
		super();
		this.name = name;
		this.band = band;
		this.link = link;
	}

	// Getters & Setters
	public long getId() {
		return id;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getBand() {
		return band;
	}

	public void setBand(String band) {
		this.band = band;
	}

	public String getLink() {
		return link;
	}

	public void setLink(String link) {
		this.link = link;
	}
	
	// ToString
	@Override
	public String toString() {
		return (this.getName() + " - " + this.getBand());
	}
	
	// Equals
	@Override
	public boolean equals(Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		final Track other = (Track) obj;
		if (!Objects.equals(this.getId(), other.getId())) {
			return false;
		}
		return (Objects.equals(this.getName(), other.getName())
				&& (Objects.equals(this.getBand(), other.getBand())));
	}
	
}
